shielder/contract/lib.rs

```diff
@@ -71,10 +69,9 @@ pub mod contract {
             &mut self,
             h_note_new: Scalar,
             proof: ZkProof,
-        ) -> Result<(), ShielderError> {
+        ) -> Result<u32, ShielderError> {
```

Comment:
let's add an explanation what the return value is (applies also to `update_note`)

Reply:
added comment